c++/src/kj/one-of-test.c++

```diff
@@ -210,6 +210,24 @@ KJ_TEST("OneOf copy/move from alternative variants") {
   }
 }
 
+KJ_TEST("OneOf equality") {
+  {
+    OneOf<int, bool> a = 5;
+    OneOf<int, bool> b = 5;
+    KJ_EXPECT(a == b);
+  }
+  {
+    OneOf<int, bool> a = 5;
+    OneOf<int, bool> b = 6;
+    KJ_EXPECT(a != b);
+  }
+  {
+    OneOf<int, bool> a = 5;
+    OneOf<int, bool> b = true;
+    KJ_EXPECT(a != b);
+  }
```

Comment:
Can you add test cases for:
* Both are bool, and equal (verifies that equality is judged correctly for more than just the first type in the list).
* Both are bool, and non-equal.
* Compare int zero to bool false (should not be equal, even though C++ considers these values equal if compared directly).

Reply:
Added these test cases